src/it/java/teammates/it/sqllogic/core/NotificationsLogicIT.java

```diff
@@ -0,0 +1,70 @@
+package teammates.it.sqllogic.core;
+
+import java.time.Instant;
+import java.util.UUID;
+
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.NotificationStyle;
+import teammates.common.datatransfer.NotificationTargetUser;
+import teammates.common.exception.EntityAlreadyExistsException;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.it.test.BaseTestCaseWithSqlDatabaseAccess;
+import teammates.sqllogic.core.NotificationsLogic;
+import teammates.storage.sqlapi.NotificationsDb;
+import teammates.storage.sqlentity.Notification;
+
+/**
+ * SUT: {@link NotificationsLogic}.
+ */
+public class NotificationsLogicIT extends BaseTestCaseWithSqlDatabaseAccess {
+
+    private NotificationsLogic notificationsLogic = NotificationsLogic.inst();
+    private NotificationsDb notificationsDb = NotificationsDb.inst();
+
+    @BeforeClass
+    public void initLogicDependencies() {
+        notificationsLogic.initLogicDependencies(notificationsDb);
+    }
+
+    @Test
+    public void testUpdateNotification()
+            throws EntityAlreadyExistsException, InvalidParametersException, EntityDoesNotExistException {
+        Instant newStartTime = Instant.parse("2012-01-01T00:00:00Z");
+        Instant newEndTime = Instant.parse("2098-01-01T00:00:00Z");
+        NotificationStyle newStyle = NotificationStyle.DARK;
+        NotificationTargetUser newTargetUser = NotificationTargetUser.INSTRUCTOR;
+        String newTitle = "An updated deprecation note";
+        String newMessage = "<p>Deprecation happens in three seconds</p>";
+
+        ______TS("success: update notification that already exists");
+        Notification notification = new Notification(Instant.parse("2011-01-01T00:00:00Z"),
+                Instant.parse("2099-01-01T00:00:00Z"), NotificationStyle.DANGER, NotificationTargetUser.GENERAL,
+                "A deprecation note", "<p>Deprecation happens in three minutes</p>");
+        notificationsDb.createNotification(notification);
+
+        UUID notificationId = notification.getNotificationId();
+        Notification expectedNotification = notificationsLogic.updateNotification(notificationId, newStartTime, newEndTime,
+                newStyle, newTargetUser, newTitle, newMessage);
+
+        assertEquals(notificationId, expectedNotification.getNotificationId());
```

Comment:
Can use verifyEquals here, makes test more readable (defined for unit tests, not sure if defined for IT as well). 

It converts to JSON and compares the string, so all fields are compared (doesn't use the class equals method to compare).

Reply:
Using verifyEquals means we need to construct another notification entity to compare which seems a bit odd. The below one I agree can use verifyEquals.

Reply:
Thanks for the catch, you are right on this.